intel_extension_for_transformers/llm/runtime/graph/models/baichuan/baichuan_utils.cpp

```diff
@@ -0,0 +1,192 @@
+  n_embd = hparams.n_embd;
+  n_vocab = hparams.n_vocab;
+  n_layer = hparams.n_layer;
+  scratch = baichuan_mem_req(n_layer);
+  model.scratchs = scratch;
+}
+
+#define MODEL_BACKEND_OFFLOAD NE_BACKEND_CPU
+void BAICHUAN::load(model_context& lctx, model_progress_callback progress_callback, void* progress_callback_user_data) {
+  auto& model = lctx.model;
+  auto& ctx = model.ctx;
+
+  size_t ctx_size;
+  size_t mmapped_size;
+  ml->calc_sizes(&ctx_size, &mmapped_size);
+  ctx_size = ctx_size * 2;
+  fprintf(stderr, "%s: ne ctx size = %7.2f MB\n", __func__, ctx_size / 1024.0 / 1024.0);
+
+  const auto& hparams = model.hparams;
+  const int head_dim = n_embd / hparams.n_head;
+  const int kv_heads = hparams.n_head;  // 1 if MQA else hparams.n_head
+  const int kv_dim = kv_heads * head_dim;
+
+  // create the ne context
+  lctx.model.buf.resize(ctx_size);
+  if (use_mlock) {
+    lctx.model.mlock_buf.init(lctx.model.buf.addr);
+    lctx.model.mlock_buf.grow_to(lctx.model.buf.size);
+  }
+
+  struct ne_init_params params = {
+      /*.mem_size   =*/lctx.model.buf.size,
+      /*.mem_buffer =*/lctx.model.buf.addr,
+      /*.no_alloc   =*/ml->use_mmap,
+  };
+
+  model.ctx = ne_init(params);
+  if (!model.ctx) {
+    throw format("ne_init() failed");
+  }
+
+  ml->ne_ctx = ctx;
+
+  model.others[0] = ml->get_tensor("model.embed_tokens.weight", {n_embd, n_vocab}, NE_BACKEND_CPU);
+  model.others[1] = ml->get_tensor("model.norm.weight", {n_embd}, NE_BACKEND_CPU);
+  model.others[2] = ml->get_tensor("lm_head.weight", {n_embd, n_vocab}, NE_BACKEND_CPU);
+  const int i_gpu_start = n_layer - n_gpu_layer;
+
+  model.layers.resize(n_layer);
+  size_t vram_total = 0;
+  for (uint32_t i = 0; i < n_layer; ++i) {
+    const ne_backend backend = int(i) < i_gpu_start ? NE_BACKEND_CPU : MODEL_BACKEND_OFFLOAD;
+    auto& layer = model.layers[i];
+    std::string layers_i = "model.layers." + std::to_string(i);
+    layer.norm[0] = ml->get_tensor(layers_i + ".input_layernorm.weight", {n_embd}, backend);
+
+    // qkv GEMM
+    layer.attn[0] = ml->get_tensor(layers_i + ".self_attn.W_pack.weight", {n_embd, 3 * n_embd}, backend);
+    layer.attn[1] = ml->get_tensor(layers_i + ".self_attn.o_proj.weight", {n_embd, n_embd}, backend);
+
+    layer.norm[1] = ml->get_tensor(layers_i + ".post_attention_layernorm.weight", {n_embd}, backend);
+
+    // ffn GEMM
+    layer.ffn[0] = ml->get_tensor(layers_i + ".mlp.gate_proj.weight",
+                                  {n_embd, uint32_t(model.hparams.inner_hidden_size)}, backend);
+    layer.ffn[1] =
+        ml->get_tensor(layers_i + ".mlp.up_proj.weight", {n_embd, uint32_t(model.hparams.inner_hidden_size)}, backend);
+    layer.ffn[2] = ml->get_tensor(layers_i + ".mlp.down_proj.weight",
+                                  {uint32_t(model.hparams.inner_hidden_size), n_embd}, backend);
+
+    layer.k_cache = d_ne_new_tensor_3d(model.ctx, NE_TYPE_F16, 5120 / 40, 4096, 40);  // [n_head, maxlen, head_size]
```

Comment:
if these are static number, set it as
```suggestion
   static const size_t n_head = 128;
   static const size_t max_len = 4096;
   static const size_t head_size = 40;
   layer.k_cache = d_ne_new_tensor_3d(model.ctx, NE_TYPE_F16, n_head, maxlen, head_size);  // [n_head, maxlen, head_size]
```

Reply:
Fixed. https://github.com/intel/intel-extension-for-transformers/pull/376/commits/f229a212e552141ddea739ed47ef6b7133931e68